src/components/TodoList/TodoList.tsx

```diff
@@ -1,44 +1,136 @@
 import React from 'react';
 import './TodoList.scss';
 
-export const TodoList: React.FC = () => (
-  <div className="TodoList">
-    <h2>Todos:</h2>
-
-    <div className="TodoList__list-container">
-      <ul className="TodoList__list">
-        <li className="TodoList__item TodoList__item--unchecked">
-          <label>
-            <input type="checkbox" readOnly />
-            <p>delectus aut autem</p>
-          </label>
-
-          <button
-            className="
-              TodoList__user-button
-              TodoList__user-button--selected
-              button
-            "
-            type="button"
-          >
-            User&nbsp;#1
-          </button>
-        </li>
-
-        <li className="TodoList__item TodoList__item--checked">
-          <label>
-            <input type="checkbox" checked readOnly />
-            <p>distinctio vitae autem nihil ut molestias quo</p>
-          </label>
-
-          <button
-            className="TodoList__user-button button"
-            type="button"
-          >
-            User&nbsp;#2
-          </button>
-        </li>
-      </ul>
-    </div>
-  </div>
-);
+type Props = {
+  todos: Todo[],
+  selectedUserId: number,
+  selectedUser: (id: number) => void
+};
+
+type State = {
+  visibleTodos: Todo[],
+  query: string;
+};
+
+export class TodoList extends React.Component<Props, State> {
+  state = {
+    visibleTodos: this.props.todos,
+    query: '',
+  };
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.todos !== this.props.todos) {
+      this.getVisible();
+    }
+  }
+
+  getVisible() {
+    this.setState({ visibleTodos: this.props.todos });
+  }
+
+  handleSelectChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const { value } = event.target;
+
+    switch (value) {
+      case ('active'):
+        this.setState(({
+          visibleTodos: this.props.todos.filter(todo => !todo.completed),
+        }));
```

Comment:
You can make method for this, to not repeat yourself in next cases and just pass filtering callback to it